x-pack/plugins/infra/public/components/data_search_error_callout.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiCallOut } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React from 'react';
+import {
+  AbortedRequestSearchStrategyError,
+  GenericSearchStrategyError,
+  SearchStrategyError,
+  ShardFailureSearchStrategyError,
+} from '../../common/search_strategies/common/errors';
+
+export const DataSearchErrorCallout: React.FC<{
+  title: React.ReactNode;
+  errors: SearchStrategyError[];
+  onRetry?: () => void;
+}> = ({ errors, onRetry, title }) => {
+  const calloutColor = errors.some((error) => error.type !== 'aborted') ? 'danger' : 'warning';
+
+  return (
+    <EuiCallOut color={calloutColor} iconType="alert" title={title}>
+      {errors?.map((error, errorIndex) => (
+        <DataSearchErrorMessage key={errorIndex} error={error} />
+      ))}
+      {onRetry ? (
+        <EuiButton color={calloutColor} size="s" onClick={onRetry}>
+          <FormattedMessage
+            id="xpack.infra.dataSearch.loadingErrorRetryButtonLabel"
+            defaultMessage="Retry"
+          />
+        </EuiButton>
+      ) : null}
+    </EuiCallOut>
+  );
+};
+
+const DataSearchErrorMessage: React.FC<{ error: SearchStrategyError }> = ({ error }) => {
+  if (error.type === 'aborted') {
+    return <AbortedRequestErrorMessage error={error} />;
+  } else if (error.type === 'generic') {
+    return <GenericErrorMessage error={error} />;
+  } else if (error.type === 'shardFailure') {
+    return <ShardFailureErrorMessage error={error} />;
+  }
+  return <p>{`${error}`}</p>;
```

Comment:
Good point, I'll try to come up with something more robust.

Reply:
To be honest I don't think this will ever hit since the `if` chain above is covering all types. If we cannot predict how an error object would look like maybe a `JSON.stringify()` is enough to make it readable.